decentraland/unity-explorer
# fix: infinite loading screens

## What does this PR change?

1. Refactor the loading screen actions, so we can easily detect which step has failed
2. Adds a Timeout for Livekit actions since it seemed that they were holding the loading screen. The timeout can be adjusted further
3. Streamlines all `GoTo` commands through the chat
4. Makes the loading screen dissapear if you are on a memory full state. The scene wont be able to load, so we need to hide it

Still TODO

1. If something fails, we should rollback the realm change. Right now it take syou to an unstable state
2. If memory is full, we should warn the user that the scene may nto be complete
3. Still the livekit issues are there, we need to debug why it cannot stop/reconnect

## How to test the changes?


1. Launch the explorer
6. Teleport around and see if you fall into deadlock (eternal loading screen)

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
